### PR TITLE
M3-1089 Cannot read property 'data' of undefined.

### PR DESCRIPTION
**Pull request by rjstires**
## Purpose
Safe access response errors on LinodeSettingsAlertsPanel.

### CODE REVIEW

File: src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsAlertsPanel.tsx [hunk 1]
```diff
@@ -231,7 +231,9 @@ class LinodeSettingsAlertsPanel extends React.Component<CombinedProps, State> {
         ));
       })
       .catch((error) => {
-        this.setState(set(lensPath(['errors']), error.response.data.errors));
+        this.setState({
+          errors: pathOr([{ reason: 'Unable to update alerts thresholds.' }], ['response', 'data', 'errors'], error)
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsAlertsPanel.tsx]: need to reset loading state here

File: src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsAlertsPanel.tsx [hunk 2]
```diff
@@ -246,7 +248,7 @@ class LinodeSettingsAlertsPanel extends React.Component<CombinedProps, State> {
       >
         {
           alertSections.map((p, idx) =>
-          <AlertSection updateFor={[p.state, p.value]} key={idx} {...p} />)
+            <AlertSection updateFor={[p.state, p.value]} key={idx} {...p} />)
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsAlertsPanel.tsx]: needs `updateFor={[this.state.errors]}`

Also there's no visible error for errors with no `field`

**Comment by rjstires** [comment]
Good catch @martinmckenna, using the mock error function here's the fixed state;
image

**Comment by patthiel** [comment]
Tested via chrome dev tools, blocked the update notifications requests. Working as expected 👍

**Pull request merged**